src/librustc_driver/lib.rs

```diff
@@ -360,21 +361,71 @@ fn handle_explain(code: &str,
     };
     match descriptions.find_description(&normalised) {
         Some(ref description) => {
-            // Slice off the leading newline and print.
-            print!("{}", &(&description[1..]).split("\n").map(|x| {
+            let output = &(&description[1..]).split("\n").map(|x| {
                 format!("{}\n", if x.starts_with("```") {
                     "```"
                 } else {
                     x
                 })
-            }).collect::<String>());
+            }).collect::<String>();
+
+            show_content_with_pager(output);
+
         }
         None => {
             early_error(output, &format!("no extended information for {}", code));
         }
     }
 }
 
+#[cfg(not(windows))]
+fn spawn_pager_with_file(file_path: &PathBuf) -> io::Result<Output> {
+    use std::os::unix::io::FromRawFd;
+    let raw_file_descriptor_id = 2; // stdout
+
+    let pager = env::var("PAGER").unwrap_or(String::from("less"));
+
+    unsafe {
+        Command::new(pager)
+            .arg(file_path)
+            .stdout(Stdio::from_raw_fd(raw_file_descriptor_id))
+            .output()
+    }
+}
+
+#[cfg(windows)]
+fn spawn_pager_with_file(file_path: &PathBuf) -> io::Result<Output> {
+    let pager = env::var("PAGER").unwrap_or(String::from("more.com"));
+
+    Command::new(pager)
+        .arg(file_path)
+        .output()
```

Comment:
I think the reason this is failing on Windows is the use of `output()` here. `output()` captures stdout which we definitely don't want, it should go to the console so using `status()` might be better. However as `more.com` and `less` accept input over stdin you should be able to do something like the following and avoid creating a file:

```rust
let mut child = Command::new(pager).stdin(Stdio::piped()).spawn()?;
child.stdin.as_mut().unwrap().write_all(content.as_bytes())?;
child.wait()
```